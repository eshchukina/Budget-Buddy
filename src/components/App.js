import React, { useState, useEffect } from "react";
import Header from "./Header";
import SideMenu from "./SideMenu";
import Dashboard from "./Dashboard";
import Footer from "./Footer";


import "./Style.css";

function App() {
  const [isDarkMode, setIsDarkMode] = useState(() => {
    // Get the theme preference from localStorage
    const storedTheme = localStorage.getItem("isDarkMode");
    // Return the stored theme if available, or set the initial state to false (light mode)
    return storedTheme ? JSON.parse(storedTheme) : false;
  });
  const [accounts, setAccounts] = useState([]);
  const [currency, setCurrency] = useState("USD");
  const [activeAccount, setActiveAccount] = useState(
    accounts.length > 0 ? accounts[0] : null
  );

  const setActiveModal = (modal) => {};

  const headersWithToken = {};

  useEffect(() => {
    if (isDarkMode) {
      document.body.classList.add("dark");
    } else {
      document.body.classList.remove("dark");
    }
    localStorage.setItem("isDarkMode", JSON.stringify(isDarkMode));
  }, [isDarkMode]);

  const toggleTheme = () => {
    setIsDarkMode((prevMode) => !prevMode);
  };


  const onAccountUpdate = (account) => {};

  const createAccount = (newAccount, currency) => {
    const account = {
      name: newAccount,
      id: accounts.length + 1,
      submittedDataList: [],
      currency: currency,
    };
    setAccounts([...accounts, account]);
    setActiveAccount(account);
  };

  const updateAccountData = (accountId, newData) => {
    setAccounts((prevAccounts) =>
      prevAccounts.map((account) =>
        account.id === accountId
          ? { ...account, submittedDataList: [...newData] }
          : account
      )
    );
  };
  const updateAccountCaption = (account) => {
    setActiveAccount((prevActiveAccount) =>
      prevActiveAccount.id === account.id
        ? { ...prevActiveAccount, ...account }
        : prevActiveAccount
    );

    setAccounts((prevAccounts) =>
      prevAccounts.map((acc) =>
        acc.id === account.id ? { ...acc, ...account } : acc
      )
    );
  };
  const handleCurrencyChange = (e) => {
    setCurrency(e.target.value);
  };

  const handleDeleteAccount = (account) => {
    const updatedAccounts = accounts.filter((acc) => acc.id !== account.id);
    setAccounts(updatedAccounts);
    setActiveAccount(updatedAccounts.length > 0 ? updatedAccounts[0] : null);
  };

  const handleDelete = (accountId, id) => {
    const updatedDataList = accounts
      .find((account) => account.id === accountId)
      .submittedDataList.filter((data) => data.id !== id);
    updateAccountData(accountId, updatedDataList);
  };



  const handleLogout = () => {
    // Удалите все данные из localStorage, связанные с авторизацией (accessToken, refreshToken и т. д.)

    localStorage.removeItem("isLoggedIn");
    localStorage.removeItem("accessToken");
    localStorage.removeItem("refreshToken");
    localStorage.removeItem("expiresIn");
  
    // Здесь также можно добавить другие действия, связанные с выходом из аккаунта, если необходимо.
  
    // Перезагрузите страницу для очистки состояния приложения.
    window.location.reload();
  };



  return (
    <div className={isDarkMode ? "dark" : "light"}>
   
            {activeAccount ? (
              <div>
        <Dashboard
          isDarkMode={isDarkMode}
          account={activeAccount}
          updateAccountData={updateAccountData}
          currency={currency}
          handleDelete={handleDelete}
          submittedDataList={activeAccount.submittedDataList}
          headersWithToken={headersWithToken}
          setActiveAccount={setActiveAccount}
          updateAccountCaption={updateAccountCaption}
          handleCurrencyChange={handleCurrencyChange}
          // dataList={dataList}

          
          createAccount={createAccount}
        
          setAccounts={setAccounts}
          accountList={accounts}
          activeAccount={activeAccount}
        
     
          handleDeleteAccount={handleDeleteAccount}
       
          setActiveModal={setActiveModal}
          onAccountUpdate={onAccountUpdate}
        
        /> 
        </div>
      ) 
      
      
      
      : (
        <div className={`mainField ${isDarkMode ? "dark" : "light"}`}>
          <div className={`text ${isDarkMode ? "dark" : "light"}`}>
            Budget Buddy is your reliable companion for efficient financial
            management! No matter where you're heading or in which currency you
            conduct your transactions, our application provides you with all the
            necessary tools to control and plan your finances. With Budget
            Buddy, you can create multiple accounts with different currencies
            that reflect your financial flows. For example, you can have an
            account for regular expenses in your local currency, as well as
            accounts for vacations or foreign investments in another currency.
            This allows you to accurately track your expenses and income for
            each account
          </div>
        </div>
      )}
      <SideMenu
        isDarkMode={isDarkMode}
        createAccount={createAccount}
        setActiveAccount={setActiveAccount}
        setAccounts={setAccounts}
        accountList={accounts}
        activeAccount={activeAccount}
        currency={currency}
        handleCurrencyChange={handleCurrencyChange}
        handleDeleteAccount={handleDeleteAccount}
        updateAccountCaption={updateAccountCaption}
        headersWithToken={headersWithToken}
        setActiveModal={setActiveModal}
        onAccountUpdate={onAccountUpdate}
        handleLogout={handleLogout}
      
      />
<Header
        isDarkMode={isDarkMode}
        toggleTheme={toggleTheme}
        handleLogout={handleLogout}
     
      />
      
      <Footer isDarkMode={isDarkMode} />
     


    </div>
  );
}

export default App;
