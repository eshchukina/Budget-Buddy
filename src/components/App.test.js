import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

test('renders app component with correct styles', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
