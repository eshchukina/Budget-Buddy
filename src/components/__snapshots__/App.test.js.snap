// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app component with correct styles 1`] = `
<div>
  <div
    class="light"
  >
    <div
      class="mainField light"
    >
      <div
        class="text light"
      >
        Budget Buddy is your reliable companion for efficient financial management! No matter where you're heading or in which currency you conduct your transactions, our application provides you with all the necessary tools to control and plan your finances. With Budget Buddy, you can create multiple accounts with different currencies that reflect your financial flows. For example, you can have an account for regular expenses in your local currency, as well as accounts for vacations or foreign investments in another currency. This allows you to accurately track your expenses and income for each account.
      </div>
    </div>
    <div>
      <div
        class="sidebar light"
      >
        <button
          class="majorButton light"
        >
          Create new account
        </button>
        <div
          class="accountButtons"
        >
          <div
            class="titleList"
          >
            Your account list:
          </div>
        </div>
        <div
          class="share"
        >
          <div
            class="item"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-share-nodes "
              data-icon="share-nodes"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M352 224c53 0 96-43 96-96s-43-96-96-96s-96 43-96 96c0 4 .2 8 .7 11.9l-94.1 47C145.4 170.2 121.9 160 96 160c-53 0-96 43-96 96s43 96 96 96c25.9 0 49.4-10.2 66.6-26.9l94.1 47c-.5 3.9-.7 7.8-.7 11.9c0 53 43 96 96 96s96-43 96-96s-43-96-96-96c-25.9 0-49.4 10.2-66.6 26.9l-94.1-47c.5-3.9 .7-7.8 .7-11.9s-.2-8-.7-11.9l94.1-47C302.6 213.8 326.1 224 352 224z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            class="item"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-envelope "
              data-icon="envelope"
              data-prefix="far"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M64 112c-8.8 0-16 7.2-16 16v22.1L220.5 291.7c20.7 17 50.4 17 71.1 0L464 150.1V128c0-8.8-7.2-16-16-16H64zM48 212.2V384c0 8.8 7.2 16 16 16H448c8.8 0 16-7.2 16-16V212.2L322 328.8c-38.4 31.5-93.7 31.5-132 0L48 212.2zM0 128C0 92.7 28.7 64 64 64H448c35.3 0 64 28.7 64 64V384c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V128z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="header light"
    >
      <h1>
        <span
          class="letter"
        >
          B
        </span>
        udget 
        <span
          class="letter"
        >
          B
        </span>
        uddy
      </h1>
      <button
        class="buttonTheme light"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-sun "
          data-icon="sun"
          data-prefix="fas"
          focusable="false"
          role="img"
          style="color: rgb(188, 136, 27);"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M361.5 1.2c5 2.1 8.6 6.6 9.6 11.9L391 121l107.9 19.8c5.3 1 9.8 4.6 11.9 9.6s1.5 10.7-1.6 15.2L446.9 256l62.3 90.3c3.1 4.5 3.7 10.2 1.6 15.2s-6.6 8.6-11.9 9.6L391 391 371.1 498.9c-1 5.3-4.6 9.8-9.6 11.9s-10.7 1.5-15.2-1.6L256 446.9l-90.3 62.3c-4.5 3.1-10.2 3.7-15.2 1.6s-8.6-6.6-9.6-11.9L121 391 13.1 371.1c-5.3-1-9.8-4.6-11.9-9.6s-1.5-10.7 1.6-15.2L65.1 256 2.8 165.7c-3.1-4.5-3.7-10.2-1.6-15.2s6.6-8.6 11.9-9.6L121 121 140.9 13.1c1-5.3 4.6-9.8 9.6-11.9s10.7-1.5 15.2 1.6L256 65.1 346.3 2.8c4.5-3.1 10.2-3.7 15.2-1.6zM160 256a96 96 0 1 1 192 0 96 96 0 1 1 -192 0zm224 0a128 128 0 1 0 -256 0 128 128 0 1 0 256 0z"
            fill="currentColor"
          />
        </svg>
      </button>
      <button
        class="logIn light"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-arrow-right-to-bracket "
          data-icon="arrow-right-to-bracket"
          data-prefix="fas"
          focusable="false"
          role="img"
          style="color: rgb(188, 136, 27);"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M352 96l64 0c17.7 0 32 14.3 32 32l0 256c0 17.7-14.3 32-32 32l-64 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l64 0c53 0 96-43 96-96l0-256c0-53-43-96-96-96l-64 0c-17.7 0-32 14.3-32 32s14.3 32 32 32zm-9.4 182.6c12.5-12.5 12.5-32.8 0-45.3l-128-128c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L242.7 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l210.7 0-73.4 73.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l128-128z"
            fill="currentColor"
          />
        </svg>
      </button>
      <button
        class="SignUp light"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-user-plus "
          data-icon="user-plus"
          data-prefix="fas"
          focusable="false"
          role="img"
          style="color: rgb(188, 136, 27);"
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 128a128 128 0 1 1 256 0A128 128 0 1 1 96 128zM0 482.3C0 383.8 79.8 304 178.3 304h91.4C368.2 304 448 383.8 448 482.3c0 16.4-13.3 29.7-29.7 29.7H29.7C13.3 512 0 498.7 0 482.3zM504 312V248H440c-13.3 0-24-10.7-24-24s10.7-24 24-24h64V136c0-13.3 10.7-24 24-24s24 10.7 24 24v64h64c13.3 0 24 10.7 24 24s-10.7 24-24 24H552v64c0 13.3-10.7 24-24 24s-24-10.7-24-24z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
